publish-mqtt/src/main.rs

```diff
@@ -303,34 +287,37 @@ async fn bluetooth_connection_loop(
                 .with_context(|| std::line!().to_string())?;
         }
 
-        // Check the state of a single sensor and act on it if appropriate.
+        // Check the state of each sensor and act on it if appropriate.
         {
-            // TODO: Iterate over sensors here rather than storing next_idx in SensorState.
-            action_next_sensor(state.clone(), session.clone())
-                .await
-                .with_context(|| std::line!().to_string())?;
+            let ids: Vec<DeviceId> = state.lock().await.sensors.keys().cloned().collect();
+            for id in ids {
+                let sensor_status = state.lock().await.sensors.get(&id).map(|sensor| {
+                    log::trace!("State of {} is {:?}", sensor.name, sensor.connection_status);
+                    sensor.connection_status
+                });
+                if let Some(connection_status) = sensor_status {
+                    action_sensor(state.clone(), session, id, connection_status)
+                        .await
+                        .with_context(|| std::line!().to_string())?;
+                }
```

Comment:
```suggestion
                let connection_status = state.lock().await.sensors.get(&id).map(|sensor| {
                    log::trace!("State of {} is {:?}", sensor.name, sensor.connection_status);
                    sensor.connection_status
                }).expect("sensors cannot be deleted");
                action_sensor(state.clone(), session, id, connection_status)
                    .await
                    .with_context(|| std::line!().to_string())?;
```
seems reasonable to unwrap here